djactasauth/backends.py

```diff
@@ -60,19 +60,33 @@ def authenticate(self, username=None, password=None, **kwargs):
                 username=username, password=password, **kwargs)
 
     def _authenticate(self, username=None, password=None, **kwargs):
-        if self.is_act_as_username(username):
-            auth_username, act_as_username = username.split(self.sepchar)
-        else:
-            auth_username = act_as_username = username
-
-        auth_user = super(ActAsModelBackend, self)._authenticate(
-                username=auth_username, password=password, **kwargs)
-        if not auth_user:
-            return auth_user
-        if auth_username != act_as_username:
-            UserModel = get_user_model()
+        self.fail_unless_one_aaa_backend_is_configured()
+        assert password is not None
+        if not self.is_act_as_username(username):
+            return None
+        for backend in auth.get_backends():
+            if not isinstance(backend, ActAsBackend):
```

Comment:
you mean to store the list comprehension into a variable, and loop through that? 'coz `for backend in auth.get_backends() if not isinstance(backend, ActAsBackend):` gives a `SyntaxError`